tests/components/mold_indicator/test_sensor.py

```diff
@@ -12,284 +12,280 @@
+                "indoor_humidity_sensor": "test.indoorhumidity",
+                "calibration_factor": 2.0,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    await hass.async_start()
+
+    hass.states.async_set(
+        "test.indoortemp", STATE_UNKNOWN, {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    await hass.async_block_till_done()
+    moldind = hass.states.get("sensor.mold_indicator")
+    assert moldind
+    assert moldind.state == "unavailable"
+    assert moldind.attributes.get(ATTR_DEWPOINT) is None
+    assert moldind.attributes.get(ATTR_CRITICAL_TEMP) is None
+
+    hass.states.async_set(
+        "test.indoortemp", "30", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    hass.states.async_set(
+        "test.outdoortemp", STATE_UNKNOWN, {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    await hass.async_block_till_done()
+    moldind = hass.states.get("sensor.mold_indicator")
+    assert moldind
+    assert moldind.state == "unavailable"
+    assert moldind.attributes.get(ATTR_DEWPOINT) is None
+    assert moldind.attributes.get(ATTR_CRITICAL_TEMP) is None
+
+    hass.states.async_set(
+        "test.outdoortemp", "25", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    hass.states.async_set(
+        "test.indoorhumidity",
+        STATE_UNKNOWN,
+        {ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE},
+    )
+    await hass.async_block_till_done()
+    moldind = hass.states.get("sensor.mold_indicator")
+    assert moldind
+    assert moldind.state == "unavailable"
+    assert moldind.attributes.get(ATTR_DEWPOINT) is None
+    assert moldind.attributes.get(ATTR_CRITICAL_TEMP) is None
+
+    hass.states.async_set(
+        "test.indoorhumidity", "20", {ATTR_UNIT_OF_MEASUREMENT: PERCENTAGE}
+    )
+    await hass.async_block_till_done()
+    moldind = hass.states.get("sensor.mold_indicator")
+    assert moldind
+    assert moldind.state == "23"
+
+    dewpoint = moldind.attributes.get(ATTR_DEWPOINT)
+    assert dewpoint
+    assert dewpoint > 4.58
+    assert dewpoint < 4.59
+
+    esttemp = moldind.attributes.get(ATTR_CRITICAL_TEMP)
+    assert esttemp
+    assert esttemp == 27.5
+
+
+async def test_sensor_changed(hass):
+    """Test the sensor_changed function."""
+    assert await async_setup_component(
+        hass,
+        sensor.DOMAIN,
+        {
+            "sensor": {
+                "platform": "mold_indicator",
+                "indoor_temp_sensor": "test.indoortemp",
+                "outdoor_temp_sensor": "test.outdoortemp",
+                "indoor_humidity_sensor": "test.indoorhumidity",
+                "calibration_factor": 2.0,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    await hass.async_start()
+
+    hass.states.async_set(
+        "test.indoortemp", "30", {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get("sensor.mold_indicator").state == "90"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)